db/migrate/20170629151616_add_unique_index_to_repos.rb

```diff
@@ -0,0 +1,5 @@
+class AddUniqueIndexToRepos < ActiveRecord::Migration[5.1]
+  def change
+    add_index :repos, [:name, :user_name], :unique => true
```

Comment:
You should add a separate uniqueness constraint. The fact that indices happen to constrain is an impl detail.